mne/decoding/search_light.py

```diff
@@ -338,7 +358,10 @@ def _sl_score(estimators, X, y):
     """
     n_iter = X.shape[-1]
     for ii, est in enumerate(estimators):
-        _score = est.score(X[..., ii], y)
+        if scoring is not None:
+            _score = scoring(est, X[..., ii], y)
```

Comment:
Ignore my last comment. Going through the code, I think error occurs because X is not 2D before being passed to `decision_function`. I can see two solutions.
* Make the input to be a 2D matrix or at least convert `X` to 2D before passing to `decision_function` 
* Follow the other API @kingjr mentioned `cross_val_score(make_pipeline(SearchLight(Ridge(), scoring=my_scorer()), X, y)`

I hope I make sense..